docs/source/release_notes/release-notes.rst

```diff
@@ -43,6 +43,11 @@ develop
          - Change
 
     *    - |fixed|
+         - ``PaxosTimestampBoundStore`` now throws ``NotCurrentLeaderException``, invalidating the timestamp store, if a bound update fails because another timestamp service on the same node proposed a smaller bound for the same sequence number.
```

Comment:
This release note is in the wrong version. We should insert a 0.42.0 header - I don't mind doing that in this PR, for the sake of expediency.